src/Microsoft.AspNet.Mvc.TagHelpers/OptionTagHelper.cs

```diff
@@ -86,7 +86,7 @@ public class OptionTagHelper : TagHelper
                     // Select this <option/> element if value attribute or content matches a selected value. Callers
                     // encode values as-needed while executing child content. But TagHelperOutput itself
                     // encodes attribute values later, when GenerateStartTag() is called.
-                    var text = await context.GetChildContentAsync();
+                    var text = (await context.GetChildContentAsync()).GetContent();
```

Comment:
Don't call `GetContent` or `Get` after CR here, do it in the next Line right when you do the `.Contains`, that way we can delay it as much as possible (may not need to call it at all if there's a value.